h/h_api/bulk_api/executor.py

```diff
@@ -0,0 +1,71 @@
+"""Implementations of an 'Executor' responsible for running bulk commands."""
+from h.h_api.bulk_api.model.report import Report
+from h.h_api.enums import CommandResult, CommandType
+
+
+class Executor:
+    """A callback to provide concrete functionality for actions in BulkJob."""
+
+    def configure(self, config):
+        """
+        Configure the job from the provided config object.
+
+        :param config: A config object
+        :return:
+        """
+        pass
+
+    def execute_batch(self, command_type, data_type, default_config, batch):
+        """
+        Execute the actions specified, returning any referenced ids.
+
+        The implementer is expected to:
+
+         * Full-fill the action
+         * Respect the configuration passed
+         * Return a list of `Report` objects in the same order as the items
+           in the batch
+
+        :param command_type: The CommandType being (e.g. UPSERT)
+        :param data_type: The DataType being modified
+        :param default_config: Configuration which applies to all actions
+        :param batch: A list of command objects
+        :return: A list of `Report` objects
+        """
+        raise NotImplementedError()
+
+    def get_items(self, data_type, ids, config):
+        """
+        Retrieve a number of items to report back to the caller.
```

Comment:
```suggestion
        Return the items identified by `ids`.
```

I think whether or not these items get reported back to the ultimate caller, is the business of whatever code is calling `get_items()`. `get_items()` itself just returns the items it's asked for

Reply:
Yup. I agree